Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Integral.lean

```diff
@@ -33,115 +33,807 @@ that the integral commutes with the continuous functional calculus under appropr
+    Continuous (fun x ↦ mkD (f x) g) := by
+  have (x) : Continuous (f x) := f_cont.comp (Continuous.prodMk_right x)
+  refine continuous_of_continuous_uncurry _ ?_
+  conv in mkD _ _ => rw [mkD_of_continuous (this x)]
+  exact f_cont
+
+-- Note: there should be a non-`mkD` version of this
+open Set in
+lemma continuousOn_mkD_of_uncurry {s : Set γ}
+    (f : γ → α → β) (g : C(α, β)) (f_cont : ContinuousOn (Function.uncurry f) (s ×ˢ univ)) :
+    ContinuousOn (fun x ↦ mkD (f x) g) s := by
+  have (x) (hx : x ∈ s) : Continuous (f x) := f_cont.comp_continuous
+    (Continuous.prodMk_right x) fun _ ↦ ⟨hx, trivial⟩
+  simp_rw [continuousOn_iff_continuous_restrict, s.restrict_def]
+  refine continuous_of_continuous_uncurry _ ?_
+  conv in mkD _ _ => rw [mkD_of_continuous (this x x.2)]
+  exact f_cont.comp_continuous (.prodMap continuous_subtype_val continuous_id)
+    fun xz ↦ ⟨xz.1.2, trivial⟩
+
+open Set in
+lemma continuous_mkD_restrict_of_uncurry {t : Set α}
+    (f : γ → α → β) (g : C(t, β)) (f_cont : ContinuousOn (Function.uncurry f) (univ ×ˢ t)) :
+    Continuous (fun x ↦ mkD (t.restrict (f x)) g) := by
+  have (x) : ContinuousOn (f x) t :=
+    f_cont.comp (Continuous.prodMk_right x).continuousOn fun _ hz ↦ ⟨trivial, hz⟩
+  refine continuous_of_continuous_uncurry _ ?_
+  conv in mkD _ _ => rw [mkD_of_continuousOn (this x)]
+  exact f_cont.comp_continuous (.prodMap continuous_id continuous_subtype_val)
+    fun xz ↦ ⟨trivial, xz.2.2⟩
+
+open Set in
+lemma continuousOn_mkD_restrict_of_uncurry {s : Set γ} {t : Set α}
+    (f : γ → α → β) (g : C(t, β))
+    (f_cont : ContinuousOn (Function.uncurry f) (s ×ˢ t)) :
+    ContinuousOn (fun x ↦ mkD (t.restrict (f x)) g) s := by
+  have (x) (hx : x ∈ s) : ContinuousOn (f x) t :=
+    f_cont.comp (Continuous.prodMk_right x).continuousOn fun _ hz ↦ ⟨hx, hz⟩
+  simp_rw [continuousOn_iff_continuous_restrict, s.restrict_def]
+  refine continuous_of_continuous_uncurry _ ?_
+  conv in mkD _ _ => rw [mkD_of_continuousOn (this x x.2)]
+  exact f_cont.comp_continuous (.prodMap continuous_subtype_val continuous_subtype_val)
+    fun xz ↦ ⟨xz.1.2, xz.2.2⟩
+
+end Continuity
+
+section Measure
+
+variable {X : Type*} [MeasurableSpace X] {μ : Measure X}
+variable {E : Type*} [NormedAddCommGroup E]
+
+-- This should probably exist for `BoundedContinuousFunction` as well...
+/-- A natural criterion for `HasFiniteIntegral` of a `C(α, E)`-valued function is the existence
+of some positive function with finite integral such that `∀ᵐ x ∂μ, ∀ z : α, ‖f x z‖ ≤ bound x`.
+
+Note that there is no dominated convergence here (hence no first-countability assumption
+on `α`). We are just using the properties of Banach-space-valued integration. -/
+lemma hasFiniteIntegral_of_bound [CompactSpace α] (f : X → C(α, E)) (bound : X → ℝ)
+    (bound_nonneg : 0 ≤ᵐ[μ] bound)
+    (bound_int : HasFiniteIntegral bound μ)
+    (bound_ge : ∀ᵐ x ∂μ, ∀ z : α, ‖f x z‖ ≤ bound x) :
+    HasFiniteIntegral f μ := by
```

Comment:
You can get away with omitting `bound_nonneg`, right? Because if `α` is empty then `HasFiniteIntegral f μ` is automatic, and if `α` is not empty, then the μ-a.e. nonnegativity of `bound` follows from `bound_ge`.